src/script/components/Modals/PrimaryModal/PrimaryModal.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useState} from 'react';
+import cx from 'classnames';
+
+import Icon from 'Components/Icon';
+import ModalComponent from 'Components/ModalComponent';
+import {initFadingScrollbar} from '../../../ui/fadingScrollbar';
+import renderElement from 'Util/renderElement';
+import {Action, PrimaryModalType} from './PrimaryModalTypes';
+import {
+  usePrimaryModalState,
+  showNextModalInQueue,
+  defaultContent,
+  removeCurrentModal,
+  addNewModalToQueue,
+} from './PrimaryModalState';
+
+export interface PrimaryModalProps {}
+
+export const PrimaryModalComponent: React.FC = () => {
+  const [inputValue, updateInputValue] = useState<string>('');
+  const [passwordValue, updatePasswordValue] = useState<string>('');
+  const [optionChecked, updateOptionChecked] = useState<boolean>(false);
+  const content = usePrimaryModalState(state => state.currentModalContent);
+  const errorMessage = usePrimaryModalState(state => state.errorMessage);
+  const updateErrorMessage = usePrimaryModalState(state => state.updateErrorMessage);
+  const updateCurrentModalContent = usePrimaryModalState(state => state.updateCurrentModalContent);
+  const currentId = usePrimaryModalState(state => state.currentModalId);
+  const isModalVisible = currentId !== null;
+  const {
+    checkboxLabel,
+    closeOnConfirm,
+    currentType,
+    inputPlaceholder,
+    messageHtml,
+    messageText,
+    modalUie,
+    onBgClick,
+    primaryAction,
+    secondaryAction,
+    titleText,
+    closeBtnTitle,
+  } = content || defaultContent;
```

Comment:
The default value should live only inside the state manager (the value it returns should be the default value and never be undefined if we don't expect it)